// tencentcloud-sdk-nodejs/tencentcloud/services/bi/v20220105/bi_client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars */
/*
 * Copyright (c) 2018 Tencent. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
const abstract_client_1 = require("../../../common/abstract_client");
/**
 * bi client
 * @class
 */
class Client extends abstract_client_1.AbstractClient {
    constructor(clientConfig) {
        super("bi.tencentcloudapi.com", "2022-01-05", clientConfig);
    }
    /**
     * 根据角色或标签查询行列权限配置
     */
    async DescribePermissionRanksInfo(req, cb) {
        return this.request("DescribePermissionRanksInfo", req, cb);
    }
    /**
     * 更新数据源
     */
    async ModifyDatasource(req, cb) {
        return this.request("ModifyDatasource", req, cb);
    }
    /**
     * 查询数据源列表
     */
    async DescribeDatasourceList(req, cb) {
        return this.request("DescribeDatasourceList", req, cb);
    }
    /**
     * 创建数据源
     */
    async CreateDatasource(req, cb) {
        return this.request("CreateDatasource", req, cb);
    }
    /**
     * 用户组数查询接口
     */
    async DescribeUserGroupTreeList(req, cb) {
        return this.request("DescribeUserGroupTreeList", req, cb);
    }
    /**
     * 创建行列权限
     */
    async CreatePermissionRanks(req, cb) {
        return this.request("CreatePermissionRanks", req, cb);
    }
    /**
     * 项目详情接口
     */
    async DescribeProjectInfo(req, cb) {
        return this.request("DescribeProjectInfo", req, cb);
    }
    /**
     * 创建项目
     */
    async CreateProject(req, cb) {
        return this.request("CreateProject", req, cb);
    }
    /**
     * DeleteUserGroupMember
     */
    async DeleteUserGroupMember(req, cb) {
        return this.request("DeleteUserGroupMember", req, cb);
    }
    /**
     * 创建云数据库
     */
    async CreateDatasourceCloud(req, cb) {
        return this.request("CreateDatasourceCloud", req, cb);
    }
    /**
     * 查询行列权限初始状态1
     */
    async DescribePermissionStatusInfo(req, cb) {
        return this.request("DescribePermissionStatusInfo", req, cb);
    }
    /**
     * 修改用户角色信息
     */
    async ModifyUserRole(req, cb) {
        return this.request("ModifyUserRole", req, cb);
    }
    /**
     * 用户组资源权限查询接口
     */
    async DescribeResourceUserGroupPageList(req, cb) {
        return this.request("DescribeResourceUserGroupPageList", req, cb);
    }
    /**
     * DeleteUserGroup
     */
    async DeleteUserGroup(req, cb) {
        return this.request("DeleteUserGroup", req, cb);
    }
    /**
     * 创建嵌出报表-强鉴权
     */
    async CreateEmbedToken(req, cb) {
        return this.request("CreateEmbedToken", req, cb);
    }
    /**
     * DescribeUserGroupInfo
     */
    async DescribeUserGroupInfo(req, cb) {
        return this.request("DescribeUserGroupInfo", req, cb);
    }
    /**
     * 创建用户角色
     */
    async CreateUserRole(req, cb) {
        return this.request("CreateUserRole", req, cb);
    }
    /**
     * 删除数据源
     */
    async DeleteDatasource(req, cb) {
        return this.request("DeleteDatasource", req, cb);
    }
    /**
     * 删除用户角色，会删除用户
     */
    async DeleteUserRole(req, cb) {
        return this.request("DeleteUserRole", req, cb);
    }
    /**
     * ModifyUserGroup
     */
    async ModifyUserGroup(req, cb) {
        return this.request("ModifyUserGroup", req, cb);
    }
    /**
     * 修改项目信息
     */
    async ModifyProject(req, cb) {
        return this.request("ModifyProject", req, cb);
    }
    /**
     * 用户角色列表
     */
    async DescribeUserRoleList(req, cb) {
        return this.request("DescribeUserRoleList", req, cb);
    }
    /**
     * CreateUserGroup
     */
    async CreateUserGroup(req, cb) {
        return this.request("CreateUserGroup", req, cb);
    }
    /**
     * 查询页面组件信息
     */
    async DescribePageWidgetList(req, cb) {
        return this.request("DescribePageWidgetList", req, cb);
    }
    /**
     * 项目-修改用户角色信息
     */
    async ModifyUserRoleProject(req, cb) {
        return this.request("ModifyUserRoleProject", req, cb);
    }
    /**
     * 修改用户角色信息
     */
    async ModifyUserDetailInfo(req, cb) {
        return this.request("ModifyUserDetailInfo", req, cb);
    }
    /**
     * CreateUserGroupMember
     */
    async CreateUserGroupMember(req, cb) {
        return this.request("CreateUserGroupMember", req, cb);
    }
    /**
     * 项目内-用户角色列表
     */
    async DescribeUserRoleProjectList(req, cb) {
        return this.request("DescribeUserRoleProjectList", req, cb);
    }
    /**
     * QueryUserGroupMember
     */
    async QueryUserGroupMember(req, cb) {
        return this.request("QueryUserGroupMember", req, cb);
    }
    /**
     * 更新云数据库
     */
    async ModifyDatasourceCloud(req, cb) {
        return this.request("ModifyDatasourceCloud", req, cb);
    }
    /**
     * 强鉴权token 清理，只有企业管理员才能调用该接口
     */
    async ClearEmbedToken(req, cb) {
        return this.request("ClearEmbedToken", req, cb);
    }
    /**
     * DescribeUserGroupMemberList
     */
    async DescribeUserGroupMemberList(req, cb) {
        return this.request("DescribeUserGroupMemberList", req, cb);
    }
    /**
     * 项目内-创建用户角色
     */
    async CreateUserRoleProject(req, cb) {
        return this.request("CreateUserRoleProject", req, cb);
    }
    /**
     * 项目内-删除用户角色
     */
    async DeleteUserRoleProject(req, cb) {
        return this.request("DeleteUserRoleProject", req, cb);
    }
    /**
     * 申请延长Token可用时间接口-强鉴权
     */
    async ApplyEmbedInterval(req, cb) {
        return this.request("ApplyEmbedInterval", req, cb);
    }
    /**
     * 行列权限项目内角色列表接口1
     */
    async DescribePermissionRoleInfo(req, cb) {
        return this.request("DescribePermissionRoleInfo", req, cb);
    }
    /**
     * 项目内-用户接口
     */
    async DescribeUserProjectList(req, cb) {
        return this.request("DescribeUserProjectList", req, cb);
    }
    /**
     * 按资源 - 更新用户组权限
     */
    async ModifyResourceUserGroupResource(req, cb) {
        return this.request("ModifyResourceUserGroupResource", req, cb);
    }
    /**
     * 页面截图导出
     */
    async ExportScreenPage(req, cb) {
        return this.request("ExportScreenPage", req, cb);
    }
    /**
     * 删除项目
     */
    async DeleteProject(req, cb) {
        return this.request("DeleteProject", req, cb);
    }
    /**
     * 更新用户组权限
     */
    async ModifyResourceUserGroup(req, cb) {
        return this.request("ModifyResourceUserGroup", req, cb);
    }
    /**
     * 项目信息
     */
    async DescribeProjectList(req, cb) {
        return this.request("DescribeProjectList", req, cb);
    }
}
exports.Client = Client;
